lib/routes/receive_payment/onchain/bitcoin_address/receive_bitcoin_address_payment_page.dart

```diff
@@ -0,0 +1,239 @@
+  final _scaffoldKey = GlobalKey<ScaffoldState>();
+
+  final _descriptionController = TextEditingController();
+  final _amountController = TextEditingController();
+  final _amountFocusNode = FocusNode();
+  var _doneAction = KeyboardDoneAction();
+
+  late OnchainPaymentLimitsResponse _onchainPaymentLimits;
+
+  PrepareReceiveResponse? prepareResponse;
+  Future<ReceivePaymentResponse>? receivePaymentResponse;
+
+  @override
+  void initState() {
+    super.initState();
+    if (_amountFocusNode.canRequestFocus) {
+      _amountFocusNode.requestFocus();
+    }
+    _doneAction = KeyboardDoneAction(focusNodes: [_amountFocusNode]);
+  }
+
+  @override
+  void dispose() {
+    _doneAction.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final texts = context.texts();
+
+    return Scaffold(
+      key: _scaffoldKey,
+      body: BlocBuilder<PaymentLimitsCubit, PaymentLimitsState>(
+        builder: (BuildContext context, PaymentLimitsState snapshot) {
+          if (snapshot.hasError) {
+            return Center(
+              child: Text(
+                texts.reverse_swap_upstream_generic_error_message(snapshot.errorMessage),
+                textAlign: TextAlign.center,
+              ),
+            );
+          }
+          if (snapshot.onchainPaymentLimits == null) {
+            final themeData = Theme.of(context);
+
+            return Center(
+              child: Loader(
+                color: themeData.primaryColor.withOpacity(0.5),
+              ),
+            );
+          }
+
+          _onchainPaymentLimits = snapshot.onchainPaymentLimits!;
+
+          return Padding(
+            padding: const EdgeInsets.only(bottom: 40.0),
+            child: SingleChildScrollView(
+              // TODO: Extract these into widgets
+              child: receivePaymentResponse == null ? _buildForm() : _buildQRCode(),
+            ),
+          );
+        },
+      ),
+      bottomNavigationBar: (receivePaymentResponse == null)
+          ? SingleButtonBottomBar(
+              stickToBottom: true,
+              text: texts.invoice_action_create,
+              onPressed: () {
+                if (_formKey.currentState?.validate() ?? false) {
+                  _createSwap();
+                }
+              },
+            )
+          : SingleButtonBottomBar(
+              stickToBottom: true,
+              text: texts.qr_code_dialog_action_close,
+              onPressed: () {
+                Navigator.of(context).pop();
+              },
+            ),
+    );
+  }
+
+  FutureBuilder<ReceivePaymentResponse> _buildQRCode() {
+    final texts = context.texts();
+    return FutureBuilder(
+      future: receivePaymentResponse,
+      builder: (BuildContext context, AsyncSnapshot<ReceivePaymentResponse> snapshot) {
+        return AddressWidget(
+          snapshot: snapshot,
+          title: texts.withdraw_funds_btc_address,
+          type: AddressWidgetType.lightning,
```

Comment:
Just checking, should this be `AddressWidgetType.bitcoin`?

Reply:
Addressed as part of: [Unify fee & limits message box widgets](https://github.com/breez/misty-breez/pull/167/commits/75bccf2f0c3639fec6a44826c0dd80367db61edd)